dll/cpl/mmsys/audio.c

```diff
@@ -174,17 +237,23 @@ AudioDlgProc(HWND hwndDlg,
              WPARAM wParam,
              LPARAM lParam)
 {
-    UNREFERENCED_PARAMETER(lParam);
-    UNREFERENCED_PARAMETER(wParam);
-    UNREFERENCED_PARAMETER(hwndDlg);
+    PGLOBAL_DATA pGlobalData;
+
+    pGlobalData = (PGLOBAL_DATA)GetWindowLongPtr(hwndDlg, DWLP_USER);
 
     switch(uMsg)
     {
         case WM_INITDIALOG:
         {
             UINT NumWavOut = waveOutGetNumDevs();
 
-            InitAudioDlg(hwndDlg);
+            pGlobalData = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, sizeof(GLOBAL_DATA));
+
+            if (!pGlobalData)
+                break;
```

Comment:
No EndDialog() call?

Reply:
Ok added it, but once it's called, the property sheet window disappears, but rundll32 is still working in background (probably with some invisible window).

I also tried `PropSheet_PressButton(GetParent(hwndDlg), PSBTN_CANCEL);` but it doesn't close window at all.

Any ideas?

Reply:
You could (a) make the allocation before starting the dialog or (b) keep the dialog working, just disable whatever functionality the allocation is responsible for.

Reply:
I'd prefer (b) way. :slightly_smiling_face:

Reply:
What happens if this fails? The data stored in the window (via DWLP_USER) is uninitialized, so that when the window proc gets called back, pGlobalData contains arbitrary data?

Reply:
Yeah right, fixed.